### PR TITLE
Make flutter run generate identical apks to flutter build apk

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
@devoncarew @jason-simmons

File: packages/flutter_tools/lib/src/commands/build_apk.dart [hunk 1]
```diff
@@ -478,7 +446,27 @@ Future<int> buildAndroid(
     return 1;
   }
 
-  if (!force && !_needsRebuild(outputFile, manifest)) {
+  Map<String, File> extraFiles = <String, File>{};
+  for (String addFile in addFiles ?? <String>[]) {
```
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: can we parse the --addFiles flag in the caller and pass the additional files as a map from APK paths to local files?

**Comment by collinjackson** [comment]
I updated this patch to remove support for the "add-file" argument since nobody seems to be using it currently.

**Comment by jason-simmons** [comment]
lgtm

**Pull request merged**